lib/gui/app/models/storage.js

```diff
@@ -0,0 +1,162 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const INDENTATION_SPACES = 2
+
+/**
+ * @summary Localstorage class and helper functions
+ * @class
+ * @public
+ */
+class Storage {
+  /**
+   * @function
+   * @public
+   *
+   * @param {String} superkey - superkey
+   *
+   * @example
+   * const potatoStorage = new Storage('potato')
+   */
+  constructor (superkey) {
+    this.superkey = superkey
+  }
+
+  /**
+   * @summary Get the whole object under the superkey
+   * @function
+   * @public
+   *
+   * @returns {Object}
+   *
+   * @example
+   * for (const key in potatoStorage.getAll()) {
+   *   console.log(key)
+   * }
+   */
+  getAll () {
+    try {
+      // JSON.parse(null) === null, so we fallback to {}
+      return JSON.parse(window.localStorage.getItem(this.superkey)) || {}
+    } catch (err) {
+      return {}
+    }
+  }
+
+  /**
+   * @summary Set the whole object under the superkey
+   * @function
+   * @public
+   *
+   * @param {Any} value - any valid JSON value
+   *
+   * @example
+   * potatoStorage.setAll({
+   *   location: 'somewhere',
+   *   freshness: 100,
+   *   edible: true
+   * })
+   */
+  setAll (value) {
+    window.localStorage.setItem(this.superkey, JSON.stringify(value, null, INDENTATION_SPACES))
+  }
+
+  /**
+   * @summary Clear the whole object under the superkey
+   * @function
+   * @public
+   *
+   * @example
+   * potatoStorage.clearAll()
+   */
+  clearAll () {
+    window.localStorage.removeItem(this.superkey)
+  }
+
+  /**
+   * @summary Get a stored value
+   * @function
+   * @public
+   *
+   * @param {String} key - object field key
+   * @param {Any} defaultValue - any valid JSON value
+   * @returns {Any} - the JSON parsed value
+   *
+   * @example
+   * potatoStorage.get('location', 'my farm')
+   */
+  get (key, defaultValue) {
+    const value = this.getAll()[key]
+
+    // eslint-disable-next-line no-undefined
+    if (value === undefined) {
+      return defaultValue
+    }
+
+    return value
+  }
+
+  /**
+   * @summary Modify a stored value
+   * @function
+   * @public
+   *
+   * @param {String} key - object field key
+   * @param {Function} func - function to apply to the value
+   * @param {Any} defaultValue - fallback value
+   * @returns {Any} - the value returned by the function applied above
+   *
+   * @example
+   * potatoStorage.modify('freshness', (freshness) => {
+   *   return freshness + 1
+   * })
+   */
+  modify (key, func, defaultValue) {
+    const obj = this.getAll()
+
+    let result = func(defaultValue)
+    // eslint-disable-next-line no-undefined
+    if (obj[key] !== undefined) {
+      result = func(obj[key])
+    }
```

Comment:
Should there be an `else` here to prevent `func` being called twice?
Also, it's not clear from the docstring that `func` will actually get called on `defaultValue`, rather than `key` being set to `defaultValue` directly.

Reply:
Good point, initially it was that way until the linter complained, but I'll find a different way around it.